plugins/azure/resoto_plugin_azure/resource/metrics.py

```diff
@@ -0,0 +1,303 @@
+from resotolib.utils import utc_str
+
+
+@define(eq=False, slots=False)
+class AzureMetricValueName:
+    kind: ClassVar[str] = "azure_metric_value_name"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "localized_value": S("localizedValue"),
+        "value": S("value"),
+    }
+    localized_value: Optional[str] = field(default=None)
+    value: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricMetadataValues:
+    kind: ClassVar[str] = "azure_metric_metadata_values"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "name": S("name") >> Bend(AzureMetricValueName.mapping),
+        "value": S("value"),
+    }
+    name: Optional[str] = field(default=None)
+    value: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricTimeSeriesValues:
+    kind: ClassVar[str] = "azure_metric_time_series_values"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "timestamp": S("timeStamp"),
+        "count": S("count"),
+        "total": S("total"),
+        "minimum": S("minimum"),
+        "maximum": S("maximum"),
+        "average": S("average"),
+    }
+    timestamp: datetime = field()
+    count: Optional[int] = field(default=None)
+    minimum: Optional[float] = field(default=None)
+    maximum: Optional[float] = field(default=None)
+    average: Optional[float] = field(default=None)
+    total: Optional[float] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricTimeSeries:
+    kind: ClassVar[str] = "azure_metric_time_series"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "metadata_values": S("metadatavalues") >> ForallBend(AzureMetricMetadataValues.mapping),
+        "data": S("data") >> ForallBend(AzureMetricTimeSeriesValues.mapping),
+    }
+    metadata_values: Optional[List[AzureMetricMetadataValues]] = field(default=None)
+    data: Optional[List[AzureMetricTimeSeriesValues]] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AzureMetricValue:
+    kind: ClassVar[str] = "azure_metric_value"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "type": S("type"),
+        "name": S("name") >> Bend(AzureMetricValueName.mapping),
+        "displayDescription": S("displayDescription"),
+        "unit": S("unit"),
+        "timeseries": S("timeseries", default=[]) >> ForallBend(AzureMetricTimeSeries.mapping),
+    }
+    id: Optional[str] = field(default=None)
+    type: Optional[str] = field(default=None)
+    name: Optional[str] = field(default=None)
+    display_description: Optional[str] = field(default=None)
+    unit: Optional[str] = field(default=None)
+    timeseries: Optional[List[AzureMetricTimeSeries]] = field(default=None)
+
+
+@define(hash=True, frozen=True)
+class AzureMetricQuery:
+    metric_name: str
+    metric_namespace: str
+    ref_id: str
+    instance_id: str
+    metric_id: str
+    aggregation: str = "average,minimum,maximum"
```

Comment:
I would opt to make this a list of strings (not concatenated). If you need it concatenated, you can derive the string easily